app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadManager.kt

```diff
@@ -327,6 +327,38 @@ class DownloadManager(
         }
     }
 
+    /**
+     * Renames manga download folder
+     *
+     * @param manga the manga
+     * @param newTitle the new manga title.
+     */
+    suspend fun renameManga(manga: Manga, newTitle: String) {
+        val source = sourceManager.getOrStub(manga.source)
+        val oldFolder = provider.findMangaDir(manga.title, source) ?: return
+        val newName = provider.getMangaDirName(newTitle)
+
+        if (oldFolder.name == newName) return
+
+        // just to be safe, don't allow downloads for this manga while renaming it
+        downloader.removeFromQueue(manga)
```

Comment:
Honestly should be added back afterwards

Reply:
I don't think the DB MangaRepository.update call is guaranteed to be done by the time this happens, so that ends up being pretty messy, I think.